homeassistant/helpers/service.py

```diff
@@ -109,3 +110,41 @@ def extract_entity_ids(hass, service_call, expand_group=True):
             return [service_ent_id]
 
         return service_ent_id
+
+
+@callback
+def async_get_state_services(hass, domain, state):
+    """Return a dict of services for domain that can be used to set state.
+
+    This function must be run in the event loop.
+    """
+    domain_services = hass.services.async_full_services().get(domain)
+
+    if not domain_services:
+        _LOGGER.warning("No services found for domain %s", domain)
+        return
+
+    services = defaultdict(dict)
+
+    for service_name, service_obj in domain_services.items():
```

Comment:
I think that this is too hacky. We should just return the 1 service that can achieve the state.

Reply:
That doesn't work with our current system of how services modify states (see https://github.com/home-assistant/home-assistant/pull/8851#discussion_r134298971).